File: lab/javascript/matematica_e_operadores/produto_de_n_numeros.js
const prompt = require("prompt-sync")({ sigint: true });

// Versão 1 — normal
function produto_de_n_numeros(n) {
  let produto = 1;
  for (let i = 0; i < n; i++) {
    let numero = parseFloat(prompt(`Digite o número ${i + 1}: `));
    produto *= numero;
  }
  return produto;
}

// Versão 2 — comprimida
function produto_de_n_numeros_comprimido(n) {
  let produto = 1;
  for (let i = 0; i < n; i++) {
    produto *= parseFloat(prompt(`Digite o número ${i + 1}: `));
  }
  return produto;
}

console.log(produto_de_n_numeros(3));
console.log(produto_de_n_numeros_comprimido(3));

File: lab/c/matematica_e_operadores/produto_de_n_numeros.c
#include <stdio.h>

// Versão 1 — normal
float produto_de_n_numeros(int n) {
    float produto = 1.0, numero;
    for (int i = 0; i < n; i++) {
        printf("Digite o número %d: ", i + 1);
        scanf("%f", &numero);
        produto *= numero;
    }
    return produto;
}

// Versão 2 — comprimida
float produto_de_n_numeros_comprimido(int n) {
    float produto = 1.0, numero;
    for (int i = 0; i < n; i++) {
        printf("Digite o número %d: ", i + 1);
        scanf("%f", &numero);
        produto *= numero;
    }
    return produto;
}

int main(void) {
    int n = 3;

    printf("%.2f\n", produto_de_n_numeros(n));
    printf("%.2f\n", produto_de_n_numeros_comprimido(n));

    return 0;
}
